### PR TITLE
Refactoring dams_object_helper to remove duplicated code. (DHH-835)

### PR DESCRIPTION
**Pull request by hweng**

### CODE REVIEW

**Comment by mcritchlow** [comment]
👏

**Comment by hweng** [comment]
@ucsdlib/developers - Please review, comment, etc... Thanks!

**Comment by VivianChu** [comment]
@hweng - Overall 👍  I'm wondering why you removed the fieldValue in getCreator function and used the empty array when your export_as_openurl function still uses that field.  Same question for getFormat function.

     def getCreator
-      # data_arr=[]
+      data_arr=[]
-      # get creator list from names:personal
-       **fieldValue=field_mapping('name_tesim')**
-      #if fieldValue != nil && fieldValue != ''
-      #  data_arr.push(fieldValue)
-      # end
-       
-      # get creator list from corporateName_tesim
-      # fieldValue=field_mapping('corporateName_tesim')
-      # if fieldValue != nil && fieldValue != ''
-      #  data_arr.push(fieldValue)
-      # end
     end
https://github.com/ucsdlib/damspas/blob/develop/app/helpers/dams_objects_helper.rb#L249

**Comment by hweng** [comment]
@VivianChu Because the mapping was long time ago before we refactored modules. And I think it need to rewrite.

**Comment by VivianChu** [comment]
:+1:

File: app/helpers/dams_objects_helper.rb [hunk 1]
```diff
@@ -612,94 +525,6 @@ def grabIcon(fileUse)
...
 	end
 
-	#---
-	# Renders a node of the COV component tree.
-	#
-	# @param index The object's component index.
-	# @return nil
-	# @author David T.
-	#---
-	def displayNode(index)
```
Comment (mcritchlow) [inline app/helpers/dams_objects_helper.rb]: just a quick clarification question. what is replacing these 3 deleted functions? `displayNode, displayComponentTree, initComponentTree`? 

I see the test suite passed ( 👍 ) but I'm wondering where the code for these ended up.

**Comment by hweng** [comment]
@mcritchlow, The following 4 methods replaced deleted methods: https://github.com/ucsdlib/damspas/blob/develop/app/helpers/dams_objects_helper.rb#L719-766

And I also checked there is nowhere calling the 3 deleted methods.

**Comment by lsitu** [comment]
👍

**Comment by hweng** [comment]
Thanks @ucsdlib/developers for reviewing. Could anybody please merge this PR? Thanks!

**Comment by hweng** [comment]
Thank you for merged it, @VivianChu

**Pull request merged**